### PR TITLE
Add game

### PR DESCRIPTION
**Pull request by Sid-Ahmed7**

### CODE REVIEW

**Review by Marwasliti — APPROVED** [review]

File: src/main/java/com/example/demo/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -19,9 +19,9 @@ public class SecurityConfiguration {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         http.csrf().disable()
+
                 .authorizeRequests()
-                .requestMatchers("/users/login", "/users/register").permitAll()
-                .anyRequest().authenticated()
+            .anyRequest().permitAll()
                 .and().httpBasic();
```
Comment (Benja-san) [inline src/main/java/com/example/demo/config/SecurityConfiguration.java:25]: configure a global permit all and securize all restricted routes

Comment (Benja-san) [reply src/main/java/com/example/demo/config/SecurityConfiguration.java:25]: saw that you securized'em with authorization annotations
it's fine then

File: src/main/java/com/example/demo/controller/GameController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private JwtService jwtService;
+
+    @PostMapping("/add")
+    public ResponseEntity<?> addGameToList(@RequestParam Long gameId,
+            @RequestHeader("Authorization") String token) {
+        token = token.substring(7);
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/GameController.java]: hardcoded number, and shift all this logic into a service (from token)

File: src/main/java/com/example/demo/controller/GameController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        Long userId = userService.findUserIdByUsername(username);
+
+        userService.addGame(userId, gameId);
+        return ResponseEntity.status(HttpStatus.OK).body(null);
+    }
+
+    @GetMapping("/{user_id}")
+    public ResponseEntity<List<GameDTO>> getListGames(@PathVariable Long user_id,
+            @RequestHeader("Authorization") String token) {
+        token = token.substring(7);
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/GameController.java]: duplicated code, shift into method from service class

File: src/main/java/com/example/demo/service/UserService.java [hunk 4]
```diff
@@ -67,4 +77,55 @@ public User updateProfilePicture(Long id, User user) {
...
+        User user = userRepository.findByUsername(username);
+        if (user != null) {
+            return user.getId();
+        } else {
+            throw new IllegalArgumentException("User not found");
+        }
+    }
+
+    public void addGame(Long userId, Long gameId) {
+        Optional<User> user = userRepository.findById(userId);
```
Comment (Benja-san) [inline src/main/java/com/example/demo/service/UserService.java:95]: uniformize methodology, 
Use Optional everywhere to manage errors or use methodology above. Select one and stick to it (communicate methodology used to group)

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/service/UserService.java [hunk 5]
```diff
@@ -121,11 +125,23 @@ private GameDTO convertToDTO(Game game) {
...
+                    .collect(Collectors.toList());
+        } else {
+            throw new IllegalArgumentException("User not found with id: " + userId);
+
+        }
+
+    }
+
+    public Long getUsernameInToken(String token) {
+        String username = jwtService.extractUsername(token.substring(7));
```
Comment (Benja-san) [inline src/main/java/com/example/demo/service/UserService.java]: why do you use this hardcoded number 7 ?

Comment (Sid-Ahmed7) [reply src/main/java/com/example/demo/service/UserService.java]: to delete the bearer in the token

File: src/main/java/com/example/demo/controller/GameController.java [hunk 6]
```diff
@@ -39,4 +41,14 @@ public ResponseEntity<List<GameDTO>> getListGames(@RequestParam Long user_id,
         List<GameDTO> games = userService.getList(user_id);
         return new ResponseEntity<>(games, HttpStatus.OK);
     }
+
+    @DeleteMapping()
+    public ResponseEntity<List<GameDTO>> deleteListGame(Long gameId,
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/GameController.java]: deleteGame ?

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**